ngraph/core/reference/src/runtime/reference/fft.cpp

```diff
@@ -546,23 +562,14 @@ namespace ngraph
                         {
                             int64_t current_fft_stride = fft_strides[axis_idx];
                             int64_t current_fft_length = fft_lengths[axis_idx];
+                            auto outer_fft_lengths =
```

Comment:
In 530 line we create the buffer. As I already said in previous comment I don't see the reason why we need to create this buffer. We don't use values from this buffer, can we remove it from the code?

Reply:
This `buffer` is used in the function `fft1d` to collect data with respect to the given axis (using the given strides) to calculate (I)FFT with respect to this axis. We can define such buffer in `fft1d`, but then we will have a frequent allocation/deallocation of the buffer. Using of the `buffer` defined in `fft` is an optimization to avoid frequent memory allocation/deallocation.

For example, such approach is used in the function `ComputeFft` from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/hlo_evaluator.cc#L930.